### PR TITLE
add python role from plat-intune-wsl

### PR DESCRIPTION
**Pull request by krilor**

### CODE REVIEW

**Review by andeki92 — APPROVED** [review]
Nice! 

Not sure if this is relevant, but I've used [mise-en-place](https://mise.jdx.dev/walkthrough.html) as an alternative to pyenv, nvm, sdkman (and for tools like helm/kubectl) for a while now, and been very happy with that. It allows specifying multiple versions and creates a 'shim' to make switching between the installed tools a lot easier. Also makes it possible to store the .tool-version (from asdf) in the repo to make it so you don't have to switch manually all the time. Thought it'd be worth a mention here

File: roles/python/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - python{{ item }}
+      - python{{ item }}-venv
+      - python{{ item }}-dev
+    update_cache: true
+  loop:
+    - "3.13" # Latest stable
+    # 3.12 is standard python3 on Ubuntu 24.04
+    - "3.9" # Default on Oracle Linux 9
+    - "3.8" # Latest available in software collection on Oracle Linux 7
+  become: true
```
Comment (andeki92) [inline roles/python/tasks/main.yml:25]: Is it not considered best-practice to install python as a non-root user?

**Comment by krilor** [comment]
> Nice!
> 
> Not sure if this is relevant, but I've used [mise-en-place](https://mise.jdx.dev/walkthrough.html) as an alternative to pyenv, nvm, sdkman (and for tools like helm/kubectl) for a while now, and been very happy with that. It allows specifying multiple versions and creates a 'shim' to make switching between the installed tools a lot easier. Also makes it possible to store the .tool-version (from asdf) in the repo to make it so you don't have to switch manually all the time. Thought it'd be worth a mention here

I must admit that I've only recently discovered mise and I have no experience with it. It looks neat. Its def on the radar.

File: roles/python/tasks/main.yml (hunk 1 above)
Comment (krilor) [reply roles/python/tasks/main.yml:25]: Yeah it's generally not advisabe to "mess" with system python as root and def not change the default.

So this is just installing python versions "on the side". So we get `python3.8` etc binaries.

**Pull request merged**